src/Interpolation/CIC.hpp

```diff
@@ -63,7 +63,135 @@ namespace ippl {
             return (interpolationWeight<GatherPoint, Index>(wlo, whi) * ...)
                    * view(interpolationIndex<GatherPoint, Index>(args)...);
         }
+        template <unsigned long ScatterPoint, unsigned long... Index, typename T, unsigned Dim,
+                  typename IndexType>
+        KOKKOS_INLINE_FUNCTION constexpr int ZigzagScatterToPoint(
+            const std::index_sequence<Index...>&,
+            const typename ippl::detail::ViewType<ippl::Vector<T, 3>, Dim>::view_type& view,
+            const Vector<T, Dim>& wlo, const Vector<T, Dim>& whi,
+            const Vector<IndexType, Dim>& args, const Vector<T, Dim>& val, T scale, const Vector<T, Dim>& hr, const NDIndex<Dim> lDom, int nghost, const Vector<T, Dim>& source) {
+            (void)hr;
+            (void)lDom;
+            (void)nghost;
+            (void)source;
+            bool isinbound = true;
+            ippl::Vector<T, Dim> depot = scale * val * (interpolationWeight<ScatterPoint, Index>(wlo, whi) * ...);
+            ippl::Vector<IndexType, Dim> index3{interpolationIndex<ScatterPoint, Index>(args)...};
+            for(unsigned int d = 0; d < Dim;d++){
+                isinbound &= (index3[d] < view.extent(d));
+            }
+            if(!isinbound){
+                return 0;
+            }
+            for(unsigned int d = 0;d < Dim;d++){
+                typename ippl::detail::ViewType<ippl::Vector<T, Dim>,
+                                            Dim>::view_type::value_type::value_type* destptr =
+                    &(view(interpolationIndex<ScatterPoint, Index>(args)...)[d]);
+                Kokkos::atomic_add(
+                    destptr,
+                    scale * val[d] * (interpolationWeight<ScatterPoint, Index>(wlo, whi) * ...));
```

Comment:
I'm not sure there is much gained in readability by having `destptr` be a separate variable instead of just putting the expression in the function call.

Reply:
you're right, probably not